### PR TITLE
virt-handler cleanup: use fakeclient

### PR DESCRIPTION
**Pull request by fossedihelm**
### What this PR does
Clean &refactor virt-handler controller unit test:
- Use fake client instead of mocking in order to simulate real behavior.

Fixes #

### Why we need it and why it was done in this way
Lower the bar for unit tests, eliminate false positives. The fake is standard for testing.
Narrow down the interface can decrease the probability of making a mistake.

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [x] PR: The PR description is expressive enough and will help future contributors
- [x] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [x] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

File: pkg/virt-handler/vm_test.go [hunk 1]
```diff
@@ -1131,13 +1130,13 @@ var _ = Describe("VirtualMachineInstance", func() {
...
-				fakeDiskChecksums := &container_disk.DiskChecksums{
+				fakeDiskChecksums := &containerdisk.DiskChecksums{
 					ContainerDiskChecksums: map[string]uint32{
 						vmi.Spec.Volumes[0].Name: uint32(1234),
 					},
-					KernelBootChecksum: container_disk.KernelBootChecksum{
-						Kernel: pointer.Uint32(33),
-						Initrd: pointer.Uint32(35),
+					KernelBootChecksum: containerdisk.KernelBootChecksum{
+						Kernel: virtpointer.P(uint32(33)),
```
Comment (jean-edouard) [inline pkg/virt-handler/vm_test.go:1140]: We usually just call it `pointer`, I don't think the name `virtpointer` is necessary now that the k8s pointers are gone.

File: pkg/virt-handler/vm_test.go [hunk 2]
```diff
@@ -1953,7 +1957,7 @@ var _ = Describe("VirtualMachineInstance", func() {
 				Threads: 1,
 			}
 
-			vmiConditions := virtcontroller.NewVirtualMachineInstanceConditionManager()
+			//vmiConditions := virtcontroller.NewVirtualMachineInstanceConditionManager()
```
Comment (jean-edouard) [inline pkg/virt-handler/vm_test.go:1960]: Dead code

**Review by jean-edouard — COMMENTED** [review]
/approve
/hold

File: pkg/virt-handler/vm_test.go [hunk 3]
```diff
@@ -875,30 +874,29 @@ var _ = Describe("VirtualMachineInstance", func() {
...
 			expectEvent(string(v1.AccessCredentialsSyncFailed), true)
+			updatedVMI, err := virtfakeClient.KubevirtV1().VirtualMachineInstances(metav1.NamespaceDefault).Get(context.TODO(), vmi.Name, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred())
+			Expect(updatedVMI.Status.Conditions).To(ConsistOf(
+				MatchFields(IgnoreExtras, Fields{
+					"Type":   Equal(v1.VirtualMachineInstanceIsMigratable),
+					"Status": Equal(k8sv1.ConditionTrue)},
+				),
+				MatchFields(IgnoreExtras, Fields{
+					"Type":   Equal(v1.VirtualMachineInstanceAccessCredentialsSynchronized),
```
Comment (xpivarc) [inline pkg/virt-handler/vm_test.go:896]: Missing the `Message` assert

File: pkg/virt-handler/vm_test.go [hunk 4]
```diff
@@ -3434,37 +3434,6 @@ var _ = Describe("CurrentMemory in Libvirt Domain", func() {
 	)
 })
 
-type vmiCondMatcher struct {
```
Comment (xpivarc) [inline pkg/virt-handler/vm_test.go:3455]: +1

File: pkg/virt-handler/vm_test.go [hunk 5]
```diff
@@ -269,6 +270,11 @@ var _ = Describe("VirtualMachineInstance", func() {
 		Expect(recorder.Events).To(BeEmpty())
 	})
 
+	addVMItoClient := func(vmi *v1.VirtualMachineInstance) {
```
[the remark this answers is not in the document]
Comment (fossedihelm) [reply pkg/virt-handler/vm_test.go]: Done

**Comment by fossedihelm** [comment]
@xpivarc Comments addressed and rebased. PTAL

**Review by xpivarc — APPROVED** [review]

**Pull request merged**